homeassistant/components/prometheus.py

```diff
@@ -0,0 +1,201 @@
+        m.labels(**self._labels(state)).set(value)
+
+    def _handle_light(self, state):
+        m = self._metric(
+            'light_state',
+            self.prometheus_client.Gauge,
+            'Load level of a light (0..1)',
+        )
+
+        try:
+            if 'brightness' in state.attributes:
+                value = state.attributes['brightness'] / 255.0
+            else:
+                value = state_helper.state_as_number(state)
+            value = value * 100
+            m.labels(**self._labels(state)).set(value)
+        except ValueError:
+            pass
+
+    def _handle_sensor(self, state):
+
+        _sensor_types = {
+            TEMP_CELSIUS: (
+                'temperature_c', self.prometheus_client.Gauge,
+                'Temperature in degrees Celsius',
+            ),
+            TEMP_FAHRENHEIT: (
+                'temperature_c', self.prometheus_client.Gauge,
+                'Temperature in degrees Celsius',
+            ),
+            '%': (
+                'relative_humidity', self.prometheus_client.Gauge,
+                'Relative humidity (0..100)',
+            ),
+            'lux': (
+                'light_lux', self.prometheus_client.Gauge,
+                'Light level in lux',
+            ),
+            'kWh': (
+                'electricity_used_kwh', self.prometheus_client.Gauge,
+                'Electricity used by this device in KWh',
+            ),
+            'V': (
+                'voltage', self.prometheus_client.Gauge,
+                'Currently reported voltage in Volts',
+            ),
+            'W': (
+                'electricity_usage_w', self.prometheus_client.Gauge,
+                'Currently reported electricity draw in Watts',
+            ),
+        }
+
+        unit = state.attributes.get('unit_of_measurement')
+        m = _sensor_types.get(unit)
+
+        if m is not None:
+            m = self._metric(*m)
+            try:
+                value = state_helper.state_as_number(state)
+                m.labels(**self._labels(state)).set(value)
+            except ValueError:
+                pass
+
+        self._battery(state)
+
+    def _handle_switch(self, state):
+        m = self._metric(
+            'switch_state',
+            self.prometheus_client.Gauge,
+            'State of the switch (0/1)',
+        )
+
+        value = state_helper.state_as_number(state)
+        m.labels(**self._labels(state)).set(value)
+
+
+class PrometheusView(HomeAssistantView):
+    url = '/api/prometheus'
+    name = 'api:prometheus'
+
+    def __init__(self, hass, prometheus_client):
+        super().__init__(hass)
+        self.prometheus_client = prometheus_client
+
+    @asyncio.coroutine
+    def get(self, request):
+        """Handle request for Prometheus metrics"""
+        _LOGGER.debug('Received Prometheus metrics request')
+
+        return web.Response(
+            body=self.prometheus_client.generate_latest(),
```

Comment:
Does this do any I/O ? You're inside a coroutine so if it does, run this command inside the executor.